File: 2108-find-first-palindromic-string-in-the-array/2108-find-first-palindromic-string-in-the-array.cpp
class Solution {
public:
    bool check(string s) {
        int i = 0, j = s.size() - 1;
        while(i<=j) {
            if(s[i] == s[j]) i++, j--;
            else return false;
        }
        return true;
    }

    string firstPalindrome(vector<string>& words) {
        for(string s : words) {
            if(check(s)) return s;
        }
        return "";
    }
};

File: 2108-find-first-palindromic-string-in-the-array/2108-find-first-palindromic-string-in-the-array.java
class Solution {
    private boolean check(String s) {
        int i = 0, j = s.length() - 1;
        while(i <= j) {
            if(s.charAt(i) == s.charAt(j)) {
                i++; j--;
            }
            else return false;
        }

        return true;
    }

    public String firstPalindrome(String[] words) {
        for(String s : words) {
            if(check(s)) return s;
        }
        
        return "";
    }
}